### ISSUE TITLE
Cannot find `OUT_DIR` when in workspace

### ISSUE THREAD

**Issue by kper** [lalrpop/lalrpop#595]
I get this error when running `cargo build` at the root of the workspace project.

```
error: environment variable `OUT_DIR` not defined
 --> foo/src/main.rs:3:1
  |
3 | lalrpop_mod!(pub calculator1); // synthesized by LALRPOP
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to previous error

```

I have a project which is a workspace. In this workspace, I have a subcrate which uses `lalrpop`. I will get the error as reported above when I try to build the project from the root. However, building directly from the subcrate works. I created a minimal example https://github.com/kper/lalrpop_workspace 

`rustc 1.50.0 (cb75ad5db 2021-02-10)`
`cargo 1.50.0 (f04e7fab7 2021-02-04)`

**Comment by pvinci** [comment]
I had a similar issue.  Your Cargo.toml is missing:
```toml
[package]
build = "build.rs"
```
and the build.rs file in the root of the project.
see https://users.rust-lang.org/t/environment-variable-out-dir-not-defined/3329

**Comment by kper** [comment]
@pvinci Thank you for letting me know :) but that didn't solve the problem.

I have done it differently. I did not need to put `build.rs` in the root folder, because I wanted to build it from a subcrate (in that case `foo` in the workspace). 

I solved it by adding the build dependency `lalrpop = "0.19.5"` to the root. By building it from the root, it would then put the generated .rs in the subfolder. Important is that `cargo` still used `build.rs` from the subcrate. Then I used the normal `mod` keyword instead of the lalrpop's macro `lalrpop_mod`.

Works for me and the grammar is still a in subcrate :)

Here is the corresponding commit https://github.com/kper/lalrpop_workspace/commit/42fba2d0091b81e5d01470ca596a6c680a60d950
